core/metacling/test/TClingTests.cxx

```diff
@@ -239,6 +239,25 @@ TEST_F(TClingTests, GetSharedLibDeps)
 }
 #endif
 
+// Check that a warning message is generated when using auto-injection.
+TEST_F(TClingTests, WarningAutoInjection)
+{
+   // Redirect stderr to a stringstream
+   std::stringstream buffer;
+   std::streambuf *old = std::cerr.rdbuf(buffer.rdbuf());
```

Comment:
gtest has a facility to capture streams `GetCapturedStderr`. I suspect we need to capture the `Stdout` as the warnings diagnostics may be printed there?

Reply:
Thank you :)
I found a usage of `GetCapturedStderr` for warnings in windows here: https://github.com/root-project/root/blob/2da8ebb89ceb6f31e92268b2d2f86db8a9d157dd/core/base/test/IncludePathTest.cxx#L25, that might fix the test.